src/include/cache.i

```diff
@@ -91,6 +91,48 @@ __wt_cache_dirty_inuse(WT_CACHE *cache)
 }
 
 /*
+ * __wt_cache_status --
+ *	Return if the cache usage exceeds the eviction or dirty targets.
+ */
+static inline void
+__wt_cache_status(WT_SESSION_IMPL *session, int *evictp, int *dirtyp)
```

Comment:
My vote is to leave the decision making up to the server i.e: keep all the logic for defining what path eviction chooses in the server, rather than sharing it between the server and the thread doing eviction.